server/src/main/java/org/elasticsearch/ResourceAlreadyUploadedException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+
+public class ResourceAlreadyUploadedException extends ElasticsearchException {
+
+    public ResourceAlreadyUploadedException(String msg, Object... args) {
+        super(msg, args);
+    }
+
+    public ResourceAlreadyUploadedException(StreamInput in) throws IOException {
```

Comment:
We could potentially add shard ID and/or primary term & gen as arguments/fields, but it's not truly necessary now.

Reply:
I agree it is not necessary. This exception is purposely thrown for control the code path. The caller is prepared to handle it and has enough information about the context.